packages/visx-wordcloud/src/useWordcloud.ts

```diff
@@ -0,0 +1,132 @@
+import { useEffect, useState } from 'react';
+import d3Cloud from 'd3-cloud';
+
+export interface BaseDatum {
+  text: string;
+}
+
+export interface WordcloudConfig<Datum extends BaseDatum> {
+  /**
+   * Width of the wordcloud layout.
+   *
+   * @default 0
+   */
+  width?: number;
+  /**
+   * Height of the wordcloud layout.
+   *
+   * @default 0
+   */
+  height?: number;
+  /**
+   * Sets the words array.
+   *
+   * @default []
+   */
+  words?: Datum[];
+  /**
+   * Sets the padding accessor function, which indicates the numerical padding for each word.
+   *
+   * @default 1
+   */
+  padding?: number | ((datum: Datum, index: number) => number);
+  /**
+   * Sets the font accessor function, which indicates the font face for each word.
+   *
+   * @default serif
+   */
+  font?: string | ((datum: Datum, index: number) => string);
+  /**
+   * Sets the fontSize accessor function, which indicates the numerical font size for each word.
+   *
+   * @default function(datum) { return Math.sqrt(datum.value); }
+   */
+  fontSize?: number | ((datum: Datum, index: number) => number);
+  /**
+   * Sets the fontStyle accessor function, which indicates the font style for each word.
+   *
+   * @default normal
+   */
+  fontStyle?: string | ((datum: Datum, index: number) => string);
+  /**
+   * Sets the fontWeight accessor function, which indicates the font weight for each word.
+   *
+   * @default normal
+   */
+  fontWeight?: string | number | ((datum: Datum, index: number) => string | number);
+  /**
+   * Sets the rotate accessor function, which indicates the rotation angle (in degrees) for each word.
+   *
+   * @default function() { return (~~(Math.random() * 6) -3) * 30; }
+   */
+  rotate?: number | ((datum: Datum, index: number) => number);
+  /**
+   * Sets the current type of spiral used for positioning words.
+   * This can either be one of the two built-in spirals, "archimedean" and "rectangular", or an arbitrary spiral generator can be used.
+   *
+   * @default archimedean
+   */
+  spiral?:
+    | 'archimedean'
+    | 'rectangular'
+    | ((size: [number, number]) => (t: number) => [number, number]);
+  /**
+   * Sets the internal random number generator, used for selecting the initial position of each word,
+   * and the clockwise/counterclockwise direction of the spiral for each word. Random function should return a number in the range [0, 1).
+   *
+   * @default Math.random
+   */
+  random?: () => number;
+}
+
+export function useWordcloud<Datum extends BaseDatum>({
+  width,
+  height,
+  font,
+  fontSize,
+  fontStyle,
+  fontWeight,
+  padding,
+  random,
+  rotate,
+  spiral,
+  words,
+}: WordcloudConfig<Datum>) {
+  const [cloudWords, setCloudWords] = useState<d3Cloud.Word[]>([]);
+
+  useEffect(() => {
```

Comment:
^I had a similar question, I think it would for sure have called the wordcloud layout for each render with the inline function. In the example specifically I think that would only have happened with a size change of the parent, but probably best to implement the example with perf in mind.

Reply:
purely for curiousity sake, do you think theres a way to change it inside the hook not to be concerned about constant rerender? i think this is unique across most of the visx packages right?

Reply:
I'm open to suggestions however I don't see how we can avoid this 🤔, I know it's not ideal but, as usual in React, it's up to the consumer to memoize or wrap fn in `useCallback`.

Reply:
Yeah I think unfortunately there's probably not a clean way to remove this, as @craciuncezar said it's common in React for memoization logic to be pushed to the consumer. We have this same requirement in other layout code in `visx`, I would guess that many consumers don't optimize their usage, but if they don't see perf issues it's probably okay (wordcloud perf may present more of an issue since it's a more complex algorithm)